### ISSUE TITLE
Podman plugin causes shell provisioners to run twice

### ISSUE THREAD

**Issue by dometto** [Polpetta/packer-plugin-podman#72]
For some reasons, when using this plugin all `shell` provisioners are run twice. Apart from being inefficient, this also causes errors (for instance, when a `shell` provisioner adds a user -- the second time it will already exists).

## Reproduce

This is the most minimal template with which I can reproduce.

`build.pkr.hcl`:

```
packer {
  required_plugins {
    podman = {
      version = ">=v0.1.0"
      source  = "github.com/polpetta/podman"
    }
  }
  required_plugins {
    docker = {
      version = "~> 1"
      source  = "github.com/hashicorp/docker"
    }
  }
}

source "podman" "test" {
  image = "ubuntu:jammy"
  commit = true
}

source "docker" "test" {
  image = "ubuntu:jammy"
  commit = true
}

build {
  sources = [
    "source.podman.test", # switch this to source.docker.test and the shell provisioner is run only once
  ]

  provisioner "shell" {
    inline = ["true"]
    inline_shebang = "/bin/sh -ex" # add -x to display script commands in packer stderr output
  }

}

```

Now run:

```
packer init -upgrade .  && packer build .
```

### Output

Notice `+ true` is run twice:

```
==> podman.test: Creating a temporary directory for sharing data...
==> podman.test: Pulling Podman image: ubuntu:jammy
    podman.test: Trying to pull docker.io/library/ubuntu:jammy...
    podman.test: Getting image source signatures
    podman.test: Copying blob sha256:a186900671ab62e1dea364788f4e84c156e1825939914cfb5a6770be2b58b4da
    podman.test: Copying config sha256:981912c48e9a89e903c89b228be977e23eeba83d42e2c8e0593a781a2b251cba
    podman.test: Writing manifest to image destination
    podman.test: 981912c48e9a89e903c89b228be977e23eeba83d42e2c8e0593a781a2b251cba
==> podman.test: Starting podman container...
    podman.test: Run command: podman run --systemd= -v /Users/3060845/.packer.d/tmp914367660:/packer-files -d -i -t --entrypoint=/bin/sh -- ubuntu:jammy
    podman.test: Container ID: 01bb1efad9c54ad68e909a3c442b57d8270a84dc932f9ec6c92220374ab1dd91
==> podman.test: Using docker communicator to connect: 10.88.0.18
==> podman.test: Provisioning with shell script: /var/folders/wh/vvkf0yhx5rbfynxqvjz6645c0000gn/T/packer-shell4166429128
==> podman.test: + true
==> podman.test: Committing the container
    podman.test: Image ID: 5ae1594dd6a425639039a1c8f7d564cf9a71ca7e4bde7d57df553be67871bbf9
==> podman.test: Provisioning with shell script: /var/folders/wh/vvkf0yhx5rbfynxqvjz6645c0000gn/T/packer-shell579271878
==> podman.test: + true
==> podman.test: Killing the container: 01bb1efad9c54ad68e909a3c442b57d8270a84dc932f9ec6c92220374ab1dd91
Build 'podman.test' finished after 4 seconds 522 milliseconds.
```

**Comment by gardar** [comment]
This should be fixed already in version `1.0.0` https://github.com/Polpetta/packer-plugin-podman/pull/42
But unfortunately you need to build it yourself due to this outstanding issue: https://github.com/Polpetta/packer-plugin-podman/issues/49#issuecomment-2210880354

**Comment by dometto** [comment]
Ah, thank you @gardar! And sorry, I had searched through closed issues for this problem, but not through closed PRs.

As you can see I've finally gotten around to using this for work. I'll have a look if I can help debug #70. Thanks for your and @Polpetta's work on this in the past year!
